### PR TITLE
Fix and normalise most identifier names in code

### PR DESCRIPTION
**Pull request by Joehuu**
This was done locally using Rider w/ ReSpeller by enabling the typo inspections and manually checking most of them. I ignored some special cases and added most of them to the dictionary. Only did identifier names for now as comments/strings are another story.

The big problem was with compound words and whether they should be open (i.e. have a space) or closed (i.e. no space). I did my best in determining based on the inspections, but ignored some like `Hitsound` because that spelling is most known to the community and the wiki page has it formatted like that too: https://osu.ppy.sh/wiki/en/Beatmapping/Hitsound.

### CODE REVIEW

File: osu.Game.Rulesets.Catch/Skinning/Default/DefaultHitExplosion.cs [hunk 1]
```diff
@@ -90,9 +90,9 @@ private void applyTransforms(int randomSeed)
                 .ResizeTo(largeFaint.Size * new Vector2(5, 1), duration, Easing.OutQuint)
                 .FadeOut(duration * 2);
 
-            const float angle_variangle = 15; // should be less than 45
-            directionalGlow1.Rotation = StatelessRNG.NextSingle(-angle_variangle, angle_variangle, randomSeed, 4);
-            directionalGlow2.Rotation = StatelessRNG.NextSingle(-angle_variangle, angle_variangle, randomSeed, 5);
+            const float angle_variable = 15; // should be less than 45
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/Skinning/Default/DefaultHitExplosion.cs:93]: should be `angle_variance`

File: osu.Game.Rulesets.Mania/UI/DefaultHitExplosion.cs [hunk 2]
```diff
@@ -42,7 +42,7 @@ public DefaultHitExplosion()
         [BackgroundDependencyLoader]
         private void load(IScrollingInfo scrollingInfo)
         {
```
Comment (peppy) [inline osu.Game.Rulesets.Mania/UI/DefaultHitExplosion.cs:44]: same here, `angle_variance`

File: osu.Game/Online/API/Requests/Responses/APIUser.cs [hunk 3]
```diff
@@ -152,10 +152,10 @@ public class UserCover
         public int ScoresRecentCount;
 
         [JsonProperty(@"beatmap_playcounts_count")]
-        public int BeatmapPlaycountsCount;
+        public int BeatmapPlayCountsCount;
 
         [JsonProperty]
```
Comment (peppy) [inline osu.Game/Online/API/Requests/Responses/APIUser.cs:157]:
```suggestion
        [JsonProperty(@"playstyle"]
```

Just to match the other ones in this file.

**Review by peppy — CHANGES_REQUESTED** [review]

**Review by peppy — APPROVED** [review]

File: osu.Game/Online/Rooms/PlaylistItem.cs [hunk 4]
```diff
@@ -113,7 +113,7 @@ public void MapObjects(IRulesetStore rulesets)
         }
 
         public bool ShouldSerializeID() => false;
-        public bool ShouldSerializeapiBeatmap() => false;
+        public bool ShouldSerializeAPIBeatmap() => false;
```
Comment (bdach) [inline osu.Game/Online/Rooms/PlaylistItem.cs:116]: Not confident about this one. This is a newtonsoft "convention" property that relies on being named the same as the `apiBeatmap` field. Need to doublecheck that this hasn't regressed.

**Comment by bdach** [comment]
Is there a reason why the relevant inspection severity level wasn't bumped up to warning/error to not have to do these sorts of pulls in the future?

File: osu.Game/Online/Rooms/PlaylistItem.cs (hunk 4 above)
Comment (peppy) [reply osu.Game/Online/Rooms/PlaylistItem.cs:116]: Missed that one. these methods need explicit comments in them.

Comment (bdach) [reply osu.Game/Online/Rooms/PlaylistItem.cs:116]: Yeah I'll get a branch going to revert this and also mark the other cases of this more clearly (not too many of these, thankfully).

**Comment by Joehuu** [comment]
I thought the remaining cases needed more thought the last time I did this (held back PRing this for weeks), but fixing them now seems simple enough. Will PR.

**Pull request merged**